framework/include/auxkernels/AuxKernel.h

```diff
@@ -94,6 +94,11 @@ class AuxKernelTempl : public MooseObject,
    */
   bool isNodal() const { return _nodal; }
 
+  /**
+   * @return whether this is a mortar auxiliary kernel
```

Comment:
what is a "mortar auxiliary kernel"?

Reply:
It would be a bit clearer if this would use `dynamic_cast<MortarNodalAuxKernelTempl<is_ad>*>(this) != nullptr`

Reply:
or if this method didn't exist and said dynamic cast was made inside the framework

Reply:
@recuero I am fine with this suggested change. Then we can make the method non-virtual

Reply:
This should be addressed now.